webapp/views/results-modal/script.js

```diff
@@ -1,16 +1,156 @@
-/* global $ */
+/* global $, get, deleteMethod */
 
-// eslint-disable-next-line no-unused-vars
-function display() {
-  const value = $('#results-select')[0].value;
-  if (value != '') {
-    $('#results-iframe').attr('src', '/output/' + value);
-  } else {
-    $('#results-iframe').attr('src', '');
+/**
+ * result modal class
+ */
+class ResultModal {
+  constructor(data = [], resultToDelete = '', currentView = 'ALL') {
+    this._data = data;
+    this.resultToDelete = resultToDelete;
+    this.currentView = currentView;
+  }
+
+  get dataAll() {
+    return this._data;
+  }
+  set dataAll(val) {
+    this._data = val;
+  }
+  get dataTimeMap() {
+    return this._data.filter(str => str.match(/^time/));
+  }
+  get dataQuery() {
+    return this._data.filter(str => str.match(/^query/));
+  }
+
+  remove(file) {
+    this.dataAll.splice(this.dataAll.indexOf(file), 1);
+  }
+
+  onClickTimeMap() {
+    const tbody = $('#results-table table tbody').empty();
+    $('#all-btn').removeClass('active');
+    $('#query-btn').removeClass('active');
+    $('#time-map-btn').addClass('active');
+
+    this.currentView = 'TIMEMAP';
+    this.dataTimeMap.forEach(file => {
+      this.appendItem(tbody, file);
+    });
+  }
```

Comment:
Do you really want to write this function three times, plus one more time for each module added?